packages/x-date-pickers-pro/src/internal/hooks/useMobileRangePicker/useMobileRangePicker.tsx

```diff
@@ -88,7 +88,7 @@ export const useMobileRangePicker = <
     externalSlotProps: componentsProps.field,
     additionalProps: {
       ...pickerFieldProps,
-      readOnly,
+      readOnly: fieldSlotsProps.startInput.readOnly,
```

Comment:
If you have better suggestions of how to not loose the props from being correctly propagated—I am open to suggestions.
The issue is that without this the `readOnly` flag is lost, while it is actually being [correctly initialized in here](https://github.com/mui/mui-x/pull/7135/files#diff-63d527e5271e05c7345f4c6abb8068bd30729006acdf288c8c66390b08b7aef3R94).

Reply:
Since you already know it's a mobile component, you could use

```suggestion
      readOnly: readOnly ?? true;
```

Reply:
Thanks for a suggestion. Aligned it with regular `useMobilePicker` by updating to `readOnly: true,`. This disables provided `readOnly` prop from being applied as we decided to not support it on mobile pickers for now.